src/module/data/message/base.mjs

```diff
@@ -15,7 +17,36 @@ export default class BaseMessageModel extends foundry.abstract.TypeDataModel {
 
   /** @inheritdoc */
   static defineSchema() {
-    return {};
+    const fields = foundry.data.fields;
+
+    return {
+      targets: new fields.SetField(new fields.DocumentUUIDField({ nullable: false }), { initial: () => Array.from(game.user.targets.map(t => t.document.uuid)) }),
+    };
+  }
+
+  /* -------------------------------------------------- */
+
+  /**
+   * Retrieves the set of tokens targeted by this message
+   * @returns {Set<DrawSteelTokenDocument>}
+   */
+  get targetTokens() {
+    return this.targets.map(uuid => fromUuidSync(uuid));
+  }
+
+  /* -------------------------------------------------- */
+
+  /**
+   * Retrieves the set of actors targeted by this message.
+   * The size of the set may be different from the targets if multiple copies of a linked token are on the scene.
+   * @returns {Set<DrawSteelActor>}
+   */
+  get targetActors() {
+    const mapped = new Set();
+    for (const tokenDoc of this.targetTokens) {
+      mapped.add(tokenDoc.actor);
+    }
+    return mapped;
```

Comment:
We have some overlap in functionality here with `ds.utils.selectedActors`. Adding an optional parameter to it (an iterable of tokens/token documents) with a fallback to `canvas.tokens.controlled` (and probably having to rename it) would work well I think.

Reply:
I think just `toActors` is fine.

Reply:
Bit too close to `fromData`. Seems to suggest it creates an actor / actors from tokens.

I think a util method is fine unless you can think of a different name.